### ISSUE TITLE
Error when running kbox provision command

### ISSUE THREAD

**Issue by yosefromano** [kalabox/kalabox#327]
Trying to update Kalabox from 0.9.2 to 0.9.3.
Got this error when running kbox provision in step 17/19:
Error running step: {"name":"engine-docker-prepared","description":"Restarting engine for updates...","deps":["core-image-prepare"],"subscribes":[],"isStep":true}: Error while shutting down.: EVENTS =>  Error dispatching event listeners [pre-down].

**Comment by pirog** [comment]
Can you run this again with verbose move turned on `kbox provision -- -v`

**Comment by yosefromano** [comment]
Got this: 
Error running step: {"name":"engine-docker-prepared","description":"Restarting engine for updates...","deps":["core-image-prepare"],"subscribes":[],"isStep":true}: Error while shutting down.: EVENTS =>  Error dispatching event listeners [pre-down].
VError: Error while shutting down.: EVENTS =>  Error dispatching event listeners [pre-down].
    at /usr/local/lib/node_modules/kalabox/node_modules/kalabox-engine-docker/provider/b2d.js:261:13
From previous event:
    at Object.down (/usr/local/lib/node_modules/kalabox/node_modules/kalabox-engine-docker/provider/b2d.js:260:11)
From previous event:
    at /usr/local/lib/node_modules/kalabox/lib/engine.js:284:31
From previous event:
    at Object.exports.down (/usr/local/lib/node_modules/kalabox/lib/engine.js:283:4)
    at step.all (/usr/local/lib/node_modules/kalabox/node_modules/kalabox-engine-docker/lib/install.js:173:21)
    at /usr/local/lib/node_modules/kalabox/lib/install/framework.js:414:17
    at /usr/local/lib/node_modules/kalabox/lib/install/framework.js:412:30
From previous event:
    at /usr/local/lib/node_modules/kalabox/lib/install/framework.js:404:29
From previous event:
    at /usr/local/lib/node_modules/kalabox/lib/install/framework.js:399:15
    at Object.exports.run (/usr/local/lib/node_modules/kalabox/lib/install.js:20:3)
    at Object.<anonymous> (/usr/local/lib/node_modules/kalabox/plugins/kalabox-core/util.js:217:23)
    at Object.exports.run (/usr/local/lib/node_modules/kalabox/lib/core/tasks.js:773:15)
    at processTask (/usr/local/lib/node_modules/kalabox/bin/kbox.js:272:16)
    at /usr/local/lib/node_modules/kalabox/bin/kbox.js:342:15
    at /usr/local/lib/node_modules/kalabox/bin/kbox.js:74:5
    at /usr/local/lib/node_modules/kalabox/node_modules/async/lib/async.js:154:25
    at processImmediate [as _immediateCallback] (timers.js:367:17)

**Comment by pirog** [comment]
@bcauldwell something to look into. @yosefromano you shuoldnt need to reprovision to go from 0.9.2 to 0.9.3, just updating the code is fine so you should be able to continue without it.

**Comment by yosefromano** [comment]
Cool thanks

**Comment by yosefromano** [comment]
I'm getting this error when trying to create an app, whether pantheon or drupal:
Error while installing app: "base": Error querying isRunning: "'31919a8cc68c405a290cd9d594e2556efc64da1a296c079f6d9ffd3d62b77973'".: Error inspecting container.: The container "'31919a8cc68c405a290cd9d594e2556efc64da1a296c079f6d9ffd3d62b77973'" does not exist!

**Comment by dsnopek** [comment]
After doing permissions wrong with #328, when I finally run `kbox provision` with the correct permissions, I'm getting this error:

```
info: -- Step 17/19 --
info: Restarting engine for updates...
info: DOCKER =>  Creating container. Image=kalabox/syncthing:v0.9.0, name=kalabox_syncthing, Binds=undefined, NetworkMode=bridge, 60008/tcp=[HostIp=, HostPort=60008], 22000/tcp=[HostIp=, HostPort=22000], 21025/udp=[HostIp=, HostPort=21025], 21026/udp=[HostIp=, HostPort=21026], VolumesFrom=[kalabox_data], Env=[KBOX_CODEDIR=code, KALABOX=true]
Error running step: {"name":"engine-docker-prepared","description":"Restarting engine for updates...","deps":["core-image-prepare"],"subscribes":[],"isStep":true}: Error while shutting down.: EVENTS =>  Error dispatching event listeners [pre-down].
```

However, just for the hell of it since @pirog said something about reprovisioning not being necessary (and I certainly tried provisioning several times when my permissions were wrong), I decided to try running `kbox create pantheon` but I get this error (after answering all the questions):

```
Error while installing app: "iklesianovember": Error querying isRunning: "'1f625d88ed0c7b5bfbaff9e33160bff9a730385fcd44ec48f99ff72c940f2ab6'".: Error inspecting container.: The container "'1f625d88ed0c7b5bfbaff9e33160bff9a730385fcd44ec48f99ff72c940f2ab6'" does not exist!
```

So, basically the same problem as @yosefromano is having...

**Comment by pirog** [comment]
@dsnopek reprovisioning shouldnt be necessary for a small version bump but since you provisioned as the root user first i think it will necessary for you to do the whole provision again. Since you might have some cruft from previous attempts there are some uninstall scripts located at in the `scripts` directory inside the kalabox code which i think gets installed in `/usr/local/lib/node_modules` by default if you do a global `npm install` on OSX/Nix

I would try running the script appropriate for your OS and then attempting another provision. Be careful NOT to uninstall VB if you use it for other stuff. Run the provision   like this `kbox provision -- -v` for verbose output.

**Comment by bcauldwell** [comment]
Looking at this issue now everyone, going to try to recreate on my machine
and then go from there.

On Thu, Jul 23, 2015 at 12:13 PM, Mike Pirog <notifications@github.com>
wrote:

> @dsnopek <https://github.com/dsnopek> reprovisioning shouldnt be
> necessary for a small version bump but since you provisioned as the root
> user first i think it will necessary for you to do the whole provision
> again. Since you might have some cruft from previous attempts there are
> some uninstall scripts located at in the scripts directory inside the
> kalabox code which i think gets installed in /usr/local/lib/node_modules
> by default if you do a global npm install on OSX/Nix
>
> I would try running the script appropriate for your OS and then attempting
> another provision. Be careful NOT to uninstall VB if you use it for other
> stuff. Run the provision like this kbox provision -- -v for verbose
> output.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/327#issuecomment-124214910>.
>



-- 
- Benjamin Cauldwell

**Comment by dsnopek** [comment]
@pirog: Deleting the ~/.kalabox directory appears to have fixed it! I ran `kbox provision`, everything went good, and now `kbox create` has gotten further than before. :-) Thanks for your help!

**Comment by bcauldwell** [comment]
I was able to recreate this error. It looks like the kalabox_syncthing
container is failing to be created because it relies on the kalabox_data
container, which for some reason after the upgrade does not exist.

I will look into this further and come up with a fix for it.

On Thu, Jul 23, 2015 at 3:18 PM, David Snopek <notifications@github.com>
wrote:

> @pirog <https://github.com/pirog>: Deleting the ~/.kalabox directory
> appears to have fixed it! I ran kbox provision, everything went good, and
> now kbox create has gotten further than before. :-) Thanks for your help!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/327#issuecomment-124253414>.
>



-- 
- Benjamin Cauldwell

**Comment by bcauldwell** [comment]
@yosefromano @dsnopek I came up with a fix and tested it. If you now update to kalabox 0.9.4 and do a provision you should be good.

`npm install kalabox@0.9.4
kbox provision`

**Comment by bcauldwell** [comment]
Upgrading to kalabox 0.9.4 will fix this issue.

On Thu, Jul 23, 2015 at 3:54 PM, Ben Cauldwell <ben@kalamuna.com> wrote:

> I was able to recreate this error. It looks like the kalabox_syncthing
> container is failing to be created because it relies on the kalabox_data
> container, which for some reason after the upgrade does not exist.
>
> I will look into this further and come up with a fix for it.
>
> On Thu, Jul 23, 2015 at 3:18 PM, David Snopek <notifications@github.com>
> wrote:
>
>> @pirog <https://github.com/pirog>: Deleting the ~/.kalabox directory
>> appears to have fixed it! I ran kbox provision, everything went good,
>> and now kbox create has gotten further than before. :-) Thanks for your
>> help!
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kalabox/kalabox/issues/327#issuecomment-124253414>.
>>
>
>
>
> --
> - Benjamin Cauldwell
>



-- 
- Benjamin Cauldwell

**Comment by pirog** [comment]
@yosefromano @dsnopek can you confirm or deny that this fix works? @bcauldwell maybe we should delete the entire `~/.kalabox` dir on uninstall?

**Comment by bcauldwell** [comment]
@pirog we could do that, however we should think about moving the instance
id file somewhere else, or if we are comfortable generating a new instance
id when someone reinstalls we can just do that.

On Fri, Jul 24, 2015 at 5:37 PM, Mike Pirog <notifications@github.com>
wrote:

> @yosefromano <https://github.com/yosefromano> @dsnopek
> <https://github.com/dsnopek> can you confirm or deny that this fix works?
> @bcauldwell <https://github.com/bcauldwell> maybe we should delete the
> entire ~/.kalabox dir on uninstall?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/327#issuecomment-124778691>.
>



-- 
- Benjamin Cauldwell

**Comment by yosefromano** [comment]
@pirog @bcauldwell I didn't get a chance to try the fix. I did like @dsnopek, I deleted /.kalabox and ran kbox provision. After that everything works!

**Comment by pirog** [comment]
@bcauldwell do we know specifically what is causing this inside of `~/.kalabox`? Do you think the fix you provided makes this workaround unneeded?

**Comment by dsnopek** [comment]
@pirog @bcauldwell Yeah, unfortunately, I don't know how to get my setup back into that state. :-/ But thanks for the super fast support!

**Comment by bcauldwell** [comment]
I really don't understand why removing ~/.kalabox should work, but the work
around does indeed work for when I was able to recreate the issue. I'd like
to explore this further down the road however when there is more time.

On Sat, Jul 25, 2015 at 1:49 PM, Mike Pirog <notifications@github.com>
wrote:

> @bcauldwell <https://github.com/bcauldwell> do we know specifically what
> is causing this inside of ~/.kalabox? Do you think the fix you provided
> makes this workaround unneeded?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/327#issuecomment-124891745>.
>



-- 
- Benjamin Cauldwell

**Comment by pirog** [comment]
@dsnopek if you've got some time maybe @bcauldwell could help you get all setup?
